src/main/java/kate/Kate.java

```diff
@@ -0,0 +1,591 @@
+            printMessage(FAILURE_MESSAGE_SET_DONE);
+        } catch (EmptyTaskException e) {
+            printMessage(FAILURE_MESSAGE_EMPTY_TASK);
+        }
+    }
+
+    /**
+     * Process user input to extract the object of the associated task number
+     *
+     * @param userInput Input provided by user
+     * @return Task object of the task that is done
+     * @throws EmptyFieldException   If task number provided is empty
+     * @throws InvalidFieldException If task number provided is invalid
+     * @throws EmptyTaskException    If task list is empty
+     */
+    private static Task processDoneInput(String userInput) throws EmptyFieldException, InvalidFieldException,
+            EmptyTaskException {
+
+        if (tasks.isEmpty()) {
+            throw new EmptyTaskException();
+        }
+
+        String[] inputArr = userInput.split(" ");
+        try {
+            String doneInput = inputArr[1];
+            if (doneInput.isEmpty()) {
+                throw new EmptyFieldException();
+            }
+
+            int taskNumber = Integer.parseInt(doneInput);
+            if ((taskNumber > tasks.size()) || (taskNumber < 1) || inputArr.length != 2) {
+                throw new InvalidFieldException();
+            }
+            int taskNumberIndex = taskNumber - 1;
+
+            return tasks.get(taskNumberIndex);
+        } catch (ArrayIndexOutOfBoundsException | NumberFormatException e) {
+            throw new InvalidFieldException();
+        }
+
+    }
+
+    /**
+     * Delete a task specified by a task number
+     *
+     * @param userInput Input provided by user
+     */
+    private static void deleteTask(String userInput) {
+        try {
+            Task deletedTask = processDeleteInput(userInput);
+            String deletedTaskInfo = deletedTask.getTaskInfo();
+            tasks.remove(deletedTask);
+
+            String deleteMessage = TEXT_INDENTATION + "Noted. I've removed this task:\n"
+                    + TEXT_INDENTATION + "  " + deletedTaskInfo + "\n"
+                    + TEXT_INDENTATION + "You currently have ("
+                    + tasks.size() + ") tasks in your list :)\n";
+            printMessage(deleteMessage);
+            updateTasksToFile();
+        } catch (EmptyFieldException | InvalidFieldException e) {
+            printMessage(FAILURE_MESSAGE_DELETE_TASK);
+        } catch (EmptyTaskException e) {
+            printMessage(FAILURE_MESSAGE_EMPTY_TASK);
+        }
+
+    }
+
+    /**
+     * Process user input to extract the deleted task number
+     *
+     * @param userInput Input provided by the user
+     * @return Task object of the deleted task
+     * @throws EmptyFieldException   If task number provided is empty
+     * @throws InvalidFieldException If task number provided is invalid
+     * @throws EmptyTaskException    If task list is empty
+     */
+    private static Task processDeleteInput(String userInput) throws EmptyFieldException, InvalidFieldException,
```

Comment:
May want to rename this since it is extracting the deleted task, similar for the others.